models/models/gan/gan_2D.py

```diff
@@ -246,6 +245,8 @@ def train(self, epochs: int | None, x_train: Any, y_train: Any, progress_callbac
             if step % batch_per_epoch == 0:
                 print('epoch: %d, discriminator_loss=%.3f,  generator_loss=%.3f \n'
                       % (epoch, d_loss[0],  g_loss[0]))
+                progress_callback([(d_loss[0], epoch), (g_loss[0], epoch)])
```

Comment:
Please also reverse the argument order:
```suggestion
                progress_callback([(epoch, d_loss[0]), (epoch, g_loss[0])])
```